src/gpu/intel/ocl/stream.cpp

```diff
@@ -136,6 +136,89 @@ status_t stream_t::barrier() {
     return impl()->barrier();
 }
 
+status_t stream_t::init_async_tracking(std::string &vinfo, double *start_ms) {
+    stream_timing_data_ = new async_timing_data_t;
+    stream_timing_data_->start_ms = get_msec();
+    stream_timing_data_->profiler_enabled = is_profiling_enabled();
+    stream_timing_data_->vinfo = vinfo;
+    return status::success;
+}
+
+status_t stream_t::register_async_tracker(cl_event event) {
+
+    // Check if there is any event in the queue - for default
+    // implementation, there is no kernel on queue to record for.
+    if (!event) return status::success;
+
+    if (!stream_timing_data_) return status::success;
+
+    cl_event exec_evt = event;
+    cl_int retain_status = clRetainEvent(exec_evt);
+    if (retain_status != CL_SUCCESS) return status::runtime_error;
+    async_tracked_event_ = exec_evt;
+
+    // local container for storing timing data for the event
+    auto event_timing_data_ = new async_timing_data_t;
+    event_timing_data_->start_ms = stream_timing_data_->start_ms;
+    event_timing_data_->vinfo = stream_timing_data_->vinfo;
+
+    event_timing_data_->profiler_enabled = is_profiling_enabled();
+
+    cl_int err = clSetEventCallback(
+            exec_evt, CL_COMPLETE, async_tracker_callback, event_timing_data_);
+    stream_timing_data_->timing_stat = (err == CL_SUCCESS);
+
+    return (stream_timing_data_->timing_stat) ? status::success
+                                              : status::runtime_error;
+}
+
+status_t stream_t::check_async_exec_times(
+        std::string &vinfo, double *start_ms) {
+
+    if (!stream_timing_data_->timing_stat) {
+        CHECK(wait());
+        double duration_ms = get_msec() - stream_timing_data_->start_ms;
+
+        VPROF(static_cast<int>(stream_timing_data_->start_ms), primitive, exec,
```

Comment:
I think we should use double as before:
```suggestion
                VPROF(stream_timing_data_->start_ms, primitive, exec,
```